Lib/test/test_compile.py

```diff
@@ -710,5 +710,102 @@ def test_func_and(self):
         self.check_stack_size(code)
 
 
+class TestStackSizeStability(unittest.TestCase):
+    # Check that repeating certain snippets doesn't increase the stack size
+    # beyond what a single snippet requires.
+
+    def check_stack_size(self, snippet):
+        sizes = []
+        for i in range(2, 5):
+            ns = {}
+            code = """def f():\n""" + i * snippet
+            code = compile(code, "<foo>", "exec")
+            exec(code, ns, ns)
+            sizes.append(ns['f'].__code__.co_stacksize)
+
+        self.assertEqual(len(set(sizes)), 1,
+                         "stack sizes diverge with # of consecutive snippets: %s" % (sizes,))
+
+    def test_if_else(self):
+        snippet = """
+            if x:
+                a
+            elif y:
+                b
+            else:
+                c
+            """
+        self.check_stack_size(snippet)
+
+    def test_try_except_bare(self):
+        snippet = """
+            try:
+                a
+            except:
+                b
+            """
+        self.check_stack_size(snippet)
+
+    def test_try_except_qualified(self):
+        snippet = """
+            try:
+                a
+            except ImportError:
+                b
+            except:
+                c
+            else:
+                d
+            """
+        self.check_stack_size(snippet)
+
+    # This one unfortunately "leaks" a few stack slots for each snippet
```

Comment:
Do you have any idea how to solve this one?

Reply:
No, I don't have not any idea. But since this test is failed with the current code I don't worry. Maybe bpo-31113 will fix it.